sklearn/pipeline.py

```diff
@@ -1630,14 +1646,23 @@ def _log_message(self, name, idx, total):
             return None
         return "(step %d of %d) Processing %s" % (idx, total, name)
 
-    def _parallel_func(self, X, y, fit_params, func):
+    def _parallel_func(self, X, y, func, fit_params):
         """Runs func in parallel on X and y"""
         self.transformer_list = list(self.transformer_list)
         self._validate_transformers()
         self._validate_transformer_weights()
         transformers = list(self._iter())
 
-        params = Bunch(fit=fit_params, fit_transform=fit_params)
+        if _routing_enabled():
+            routed_params = process_routing(self, "fit", **fit_params)
+        else:
+            # TODO(SLEP6): remove when metadata routing cannot be disabled.
+            routed_params = Bunch()
+            for name, _ in self.transformer_list:
+                routed_params[name] = Bunch(fit={})
+                routed_params[name] = Bunch(fit_transform={})
+                routed_params[name].fit = fit_params
+                routed_params[name].fit_transform = fit_params
```

Comment:
I think following the other estimators' pattern, it'd be nicer to do the routing in `fit` and `fit_transform` instead of here. And then here the method accepts `fit_params` and `transform_params` (if necessary)

Reply:
Yes, that's certainly cleaner. I've made this.
Turns out, we can use the same param (called `routed_params`) in `_parallel_func` for any routed params, no matter where they came from.